packages/client/hmi-client/src/workflow/tera-workflow.vue

```diff
@@ -281,18 +284,28 @@ function appendInputPort(
 
 function appendOutputPort(
 	node: WorkflowNode<any> | null,
-	port: { type: string; label?: string; value: any }
+	port: { type: string; label?: string; value: any; state?: any; isSelected?: boolean }
 ) {
 	if (!node) return;
 
-	node.outputs.push({
-		id: uuidv4(),
+	const uuid = uuidv4();
+
+	const outputPort: WorkflowOutput<any> = {
+		id: uuid,
 		type: port.type,
 		label: port.label,
 		value: isArray(port.value) ? port.value : [port.value],
 		isOptional: false,
-		status: WorkflowPortStatus.NOT_CONNECTED
-	});
+		status: WorkflowPortStatus.NOT_CONNECTED,
+		state: port.state,
+		timestamp: new Date()
+	};
+
+	if ('isSelected' in port) outputPort.isSelected = port.isSelected;
```

Comment:
I only want to add the isSelected key if its explicitly defined in the port, see `tera-operator-outputs` comment for more detail